components/layout/display_list_builder.rs

```diff
@@ -1024,6 +1028,194 @@ fn convert_ellipse_size_keyword(keyword: ShapeExtent,
     }
 }
 
+/// Subtract offsets from a bounding box.
+///
+/// As an example if the bounds are the border-box and the border
+/// is provided as offsets the result will be the padding-box.
+fn calculate_inner_bounds(mut bounds: Rect<Au>, offsets: SideOffsets2D<Au>) -> Rect<Au> {
+    bounds.origin.x += offsets.left;
+    bounds.origin.y += offsets.top;
+    bounds.size.width -= offsets.horizontal();
+    bounds.size.height -= offsets.vertical();
+    bounds
+}
+
+
+/// For a given area and an image compute how big the
+/// image should be displayed on the background.
+fn compute_background_image_size(bg_size: BackgroundSize<LengthOrPercentageOrAuto>,
+                                 bounds_size: Size2D<Au>,
+                                 intrinsic_size: Option<Size2D<Au>>)
+                                 -> Size2D<Au> {
+    let own_size = if let Some(size) = intrinsic_size {
+        size
+    } else {
+        return match bg_size {
```

Comment:
This may be cleaner with `match intrinsic_size { ...`, but no strong preference I guess

Reply:
I am not sure if understand this. `match bg_size { ...` matches on the CSS property `background-size`. Maybe you meant I should replace `own_size` with `intrinsic_size`?

Reply:
As a replacement for `if let Some(size) = intrinsic_size` on line 1050

Reply:
Thanks. Will change it.